### PR TITLE
Chef 15 compatibility fixes

### PR DESCRIPTION
**Pull request by detjensrobert**
Idempotency has been disabled for this cookbook as `yum_version_lock` and the `ruby_block` resources run on every converge.

### CODE REVIEW

**Comment by detjensrobert** [comment]
[Chef 15 compatibility fixes](https://trello.com/c/k2lpIE2U/1570-chef-15-compatibility-fixes)

**Comment by detjensrobert** [comment]
[Chef 15 compatibility fixes](https://trello.com/c/k2lpIE2U/1570-chef-15-compatibility-fixes)

File: Berksfile [hunk 1]
```diff
@@ -1,14 +1,17 @@
...
-cookbook 'osl-php', git: 'git@github.com:osuosl-cookbooks/osl-php'
-cookbook 'osl-selinux', git: 'git@github.com:osuosl-cookbooks/osl-selinux'
+cookbook 'munin', git: 'git@github.com:osuosl-cookbooks/munin'
+cookbook 'osl-acme', git: 'git@github.com:osuosl-cookbooks/osl-acme', branch: 'detjensrobert/chef15'
+cookbook 'osl-apache', git: 'git@github.com:osuosl-cookbooks/osl-apache', branch: 'ramereth/chef15'
+cookbook 'osl-munin', git: 'git@github.com:osuosl-cookbooks/osl-munin', branch: 'ramereth/chef15'
+cookbook 'osl-nrpe', git: 'git@github.com:osuosl-cookbooks/osl-nrpe', branch: 'ramereth/chef15'
+cookbook 'osl-rsync', git: 'git@github.com:osuosl-cookbooks/osl-rsync', branch: 'ramereth/chef15'
+cookbook 'osl-php', git: 'git@github.com:osuosl-cookbooks/osl-php', branch: 'ramereth/chef15'
+cookbook 'osl-selinux', git: 'git@github.com:osuosl-cookbooks/osl-selinux', branch: 'ramereth/chef15'
```
Comment (ramereth) [inline Berksfile]: This was merged so you can remove the branch spec.

File: Berksfile [hunk 2]
```diff
@@ -1,14 +1,17 @@
...
 cookbook 'boulder_test', path: 'test/cookbooks/boulder_test'
-cookbook 'firewall', git: 'git@github.com:osuosl-cookbooks/firewall'
-cookbook 'osl-docker', git: 'git@github.com:osuosl-cookbooks/osl-docker'
+cookbook 'firewall', git: 'git@github.com:osuosl-cookbooks/firewall', branch: 'ramereth/chef15'
```
Comment (ramereth) [inline Berksfile]: This was merged so you can remove the branch spec.

File: Berksfile [hunk 3]
```diff
@@ -1,14 +1,17 @@
...
-cookbook 'osl-nrpe', git: 'git@github.com:osuosl-cookbooks/osl-nrpe'
-cookbook 'osl-rsync', git: 'git@github.com:osuosl-cookbooks/osl-rsync'
-cookbook 'osl-php', git: 'git@github.com:osuosl-cookbooks/osl-php'
-cookbook 'osl-selinux', git: 'git@github.com:osuosl-cookbooks/osl-selinux'
+cookbook 'munin', git: 'git@github.com:osuosl-cookbooks/munin'
+cookbook 'osl-acme', git: 'git@github.com:osuosl-cookbooks/osl-acme', branch: 'detjensrobert/chef15'
+cookbook 'osl-apache', git: 'git@github.com:osuosl-cookbooks/osl-apache', branch: 'ramereth/chef15'
+cookbook 'osl-munin', git: 'git@github.com:osuosl-cookbooks/osl-munin', branch: 'ramereth/chef15'
+cookbook 'osl-nrpe', git: 'git@github.com:osuosl-cookbooks/osl-nrpe', branch: 'ramereth/chef15'
+cookbook 'osl-rsync', git: 'git@github.com:osuosl-cookbooks/osl-rsync', branch: 'ramereth/chef15'
```
Comment (ramereth) [inline Berksfile]: This was merged so you can remove the branch spec.

File: kitchen.yml [hunk 4]
```diff
@@ -1,6 +1,11 @@
 ---
 verifier:
   name: inspec
+provisioner:
+  name: chef_solo
+  # enforce_idempotency: true
+  multiple_converge: 2
```
Comment (ramereth) [inline kitchen.yml]: Please comment this out and leave a message why.

File: metadata.rb [hunk 5]
```diff
@@ -2,16 +2,13 @@
 maintainer       'Oregon State University'
 maintainer_email 'chef@osuosl.org'
 license          'Apache-2.0'
-chef_version     '>= 12.18' if respond_to?(:chef_version)
+chef_version     '>= 12.18'
```
Comment (ramereth) [inline metadata.rb]: Looks like this got missed on a previous update.
```suggestion
chef_version     '>= 14'
```

**Review by ramereth — CHANGES_REQUESTED** [review]
Can you please create a new issue on this cookbook to deal with refactoring the ``ruby_block`` resources into libraries/resources so we can properly have some idempotency with this cookbook.

**Comment by detjensrobert** [comment]
`ruby_block` removal PR: https://github.com/osuosl-cookbooks/osl-letsencrypt-boulder-server/pull/14

**Comment by ramereth** [comment]
[Chef 15: osl-letsencrypt-boulder-server](https://trello.com/c/JewZuxTp/1758-chef-15-osl-letsencrypt-boulder-server)

File: Berksfile [hunk 6]
```diff
@@ -1,14 +1,17 @@
...
 
-cookbook 'osl-acme', git: 'git@github.com:osuosl-cookbooks/osl-acme'
-cookbook 'osl-apache', git: 'git@github.com:osuosl-cookbooks/osl-apache'
-cookbook 'osl-munin', git: 'git@github.com:osuosl-cookbooks/osl-munin'
-cookbook 'osl-nrpe', git: 'git@github.com:osuosl-cookbooks/osl-nrpe'
+cookbook 'munin', git: 'git@github.com:osuosl-cookbooks/munin'
+cookbook 'osl-acme', git: 'git@github.com:osuosl-cookbooks/osl-acme', branch: 'detjensrobert/chef15'
+cookbook 'osl-apache', git: 'git@github.com:osuosl-cookbooks/osl-apache', branch: 'ramereth/chef15'
+cookbook 'osl-munin', git: 'git@github.com:osuosl-cookbooks/osl-munin', branch: 'ramereth/chef15'
+cookbook 'osl-nrpe', git: 'git@github.com:osuosl-cookbooks/osl-nrpe', branch: 'ramereth/chef15'
```
Comment (ramereth) [inline Berksfile]: These three were merged so you can update these now. Same for osl-php.

File: Berksfile [hunk 7]
```diff
@@ -1,14 +1,19 @@
+solver :ruby, :required
+
 source 'https://supermarket.chef.io'
 
-cookbook 'osl-acme', git: 'git@github.com:osuosl-cookbooks/osl-acme'
+cookbook 'base', git: 'git@github.com:osuosl-cookbooks/base'
+cookbook 'munin', git: 'git@github.com:osuosl-cookbooks/munin'
+cookbook 'osl-acme', git: 'git@github.com:osuosl-cookbooks/osl-acme
```
Comment (ramereth) [inline Berksfile]:
```suggestion
cookbook 'osl-acme', git: 'git@github.com:osuosl-cookbooks/osl-acme'
```

File: Berksfile [hunk 8]
```diff
@@ -1,14 +1,19 @@
...
 cookbook 'osl-munin', git: 'git@github.com:osuosl-cookbooks/osl-munin'
+cookbook 'osl-git', git: 'git@github.com:osuosl-cookbooks/osl-git'
 cookbook 'osl-nrpe', git: 'git@github.com:osuosl-cookbooks/osl-nrpe'
 cookbook 'osl-rsync', git: 'git@github.com:osuosl-cookbooks/osl-rsync'
 cookbook 'osl-php', git: 'git@github.com:osuosl-cookbooks/osl-php'
 cookbook 'osl-selinux', git: 'git@github.com:osuosl-cookbooks/osl-selinux'
 cookbook 'boulder_test', path: 'test/cookbooks/boulder_test'
 cookbook 'firewall', git: 'git@github.com:osuosl-cookbooks/firewall'
-cookbook 'osl-docker', git: 'git@github.com:osuosl-cookbooks/osl-docker'
+cookbook 'osl-docker', git: 'git@github.com:osuosl-cookbooks/osl-docker', branch: 'detjensrobert/chef15'
```
Comment (ramereth) [inline Berksfile:17]: Now you need to remove this branch reference :laughing:

**Review by ramereth — APPROVED** [review]
Please do a minor bump.

**Comment by detjensrobert** [comment]
!bump minor *

**Comment by osuosl-manatee** [comment]
Jenkins has merged this PR into `master` and has automatically performed a minor-level version bump to v3.2.0.  Have a nice day!

**Pull request merged**